code/scripting/api/objs/hudconfig.cpp

```diff
@@ -0,0 +1,313 @@
+{
+	if (!isValid()) {
+		return nullptr;
+	}
+
+	return &HC_gauge_regions[GR_1024][gauge];
+}
+
+int gauge_config_h::getIndex() const
+{
+	return gauge;
+}
+
+const char* gauge_config_h::getName() const
+{
+	if (!isValid()) {
+		return nullptr;
+	}
+
+	return HC_gauge_descriptions(gauge);
+}
+
+bool gauge_config_h::isValid() const
+{
+	return gauge >= 0 && gauge <= NUM_HUD_GAUGES;
+}
+
+hud_preset_h::hud_preset_h() : preset(-1) {}
+hud_preset_h::hud_preset_h(int l_preset) : preset(l_preset) {}
+
+int hud_preset_h::getIndex() const
+{
+	return preset;
+}
+
+SCP_string hud_preset_h::getName() const
+{
+	if (!isValid()) {
+		return "";
+	}
+
+	return HC_preset_filenames[preset];
+}
+
+bool hud_preset_h::isValid() const
+{
+	return preset >= 0 && preset < (int)HC_preset_filenames.size();
+}
+
+//**********HANDLE: hud preset
+ADE_OBJ(l_HUD_Preset, hud_preset_h, "hud_preset", "Hud preset handle");
+
+ADE_VIRTVAR(Name, l_HUD_Preset, nullptr, "The name of this preset", "string", "The name")
+{
+	hud_preset_h current;
+	if (!ade_get_args(L, "o", l_HUD_Preset.Get(&current))) {
+		return ADE_RETURN_NIL;
+	}
+
+	if (!current.isValid()) {
+		return ade_set_error(L, "s", "");
+	}
+
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "This property is read only.");
+	}
+
+	return ade_set_args(L, "s", current.getName().c_str());
+}
+
+ADE_FUNC(deletePreset, l_HUD_Preset, nullptr, "Deletes the preset file", nullptr, nullptr)
+{
+	hud_preset_h current;
+	if (!ade_get_args(L, "o", l_HUD_Preset.Get(&current))) {
+		return ADE_RETURN_NIL;
+	}
+
+	if (!current.isValid()) {
+		return ade_set_error(L, "s", "");
+	}
+
+	hud_config_delete_preset(current.getName());
+
+	return ADE_RETURN_NIL;
+}
+
+ADE_FUNC(isValid, l_HUD_Preset, nullptr, "Detects whether handle is valid", "boolean", "true if valid, false if handle is invalid, nil if a syntax/type error occurs")
+{
+	hud_preset_h current;
+	if (!ade_get_args(L, "o", l_HUD_Preset.Get(&current)))
+		return ADE_RETURN_NIL;
+
+	return ade_set_args(L, "b", current.isValid());
+}
+
+//**********HANDLE: gauge config
+ADE_OBJ(l_Gauge_Config, gauge_config_h, "gauge_config", "Gauge config handle");
+
+ADE_VIRTVAR(Name, l_Gauge_Config, nullptr, "The name of this gauge", "string", "The name")
+{
+	gauge_config_h current;
+	if (!ade_get_args(L, "o", l_Gauge_Config.Get(&current))) {
+		return ADE_RETURN_NIL;
+	}
+
+	if (!current.isValid()) {
+		return ade_set_error(L, "s", "");
+	}
+
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "This property is read only.");
+	}
+
+	return ade_set_args(L, "s", current.getName());
+}
+
+ADE_VIRTVAR(CurrentColor,
+	l_Gauge_Config,
+	"color",
+	"Gets the current color of the gauge. If setting the color, gauges that use IFF for color cannot be set.",
+	"color",
+	"The gauge color")
+{
+	gauge_config_h current;
+	color newColor;
+	if (!ade_get_args(L, "o|o", l_Gauge_Config.Get(&current), l_Color.Get(&newColor))) {
+		return ADE_RETURN_NIL;
+	}
+
+	if (!current.isValid()) {
+		return ade_set_error(L, "s", "");
```

Comment:
This should return a color (whether invalid or default), not a string.  The type should be consistent with the usual return type.

Reply:
how, exactly, do you expect this to return a color if the gauge isn't valid?

Reply:
Nil makes more sense. I'm doing that.

Reply:
It's not expected to be a meaningful color, it's just a placeholder color to prevent a script crash.  I used `Color_text_normal` for the mission log.  You could use white, or green, or perhaps `Color_ui_green` - whatever makes sense.

Reply:
Fine. I'll do it the way you want even though I'm the one designing the API and writing all the scripts... You do still always gotta be in control, don't ya?